src/main/java/org/skriptlang/skript/bukkit/recipes/elements/ExprRecipeGroup.java

```diff
@@ -0,0 +1,128 @@
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.*;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.recipes.MutableRecipe;
+import org.skriptlang.skript.bukkit.recipes.MutableRecipe.MutableCookingRecipe;
+import org.skriptlang.skript.bukkit.recipes.MutableRecipe.MutableCraftingRecipe;
+import org.skriptlang.skript.bukkit.recipes.MutableRecipe.MutableStonecuttingRecipe;
+import org.skriptlang.skript.bukkit.recipes.RegisterRecipeEvent;
+
+@Name("Recipe Group")
+@Description({
+	"The recipe group of a shaped, shapeless, blasting, furnace, campfire, smoking, or stonecutting recipe.",
+	"Groups recipes together under the provided string."
+})
+@Examples({
+	"set {_recipe} to a new shapeless recipe with the key \"my_recipe\":",
+		"\tset the recipe ingredients to 3 diamonds, 3 emeralds and 3 netherite ingots",
+		"\tset the recipe group to \"my group\"",
+		"\tset the recipe result to nether star"
+})
+@Since("INSERT VERSION")
+public class ExprRecipeGroup extends PropertyExpression<Recipe, String> {
+
+	static {
+		Skript.registerExpression(ExprRecipeGroup.class, String.class, ExpressionType.PROPERTY,
+			"[the] recipe group [of %recipes%]",
+			"%recipes%'[s] recipe group");
+	}
+
+	private boolean isEvent = false;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (exprs[0].isDefault()) {
+			if (exprs[0] == null) {
+				Skript.error("There is no recipe in a '" + getParser().getCurrentEventName() + "' event.");
+				return false;
+			}
+			if (getParser().isCurrentEvent(RegisterRecipeEvent.class))
+				isEvent = true;
+		}
+		//noinspection unchecked
+		setExpr((Expression<? extends Recipe>) exprs[0]);
+		return true;
+	}
+
+	@Override
+	protected String[] get(Event event, Recipe[] source) {
+		return get(source, recipe -> {
+			if (recipe instanceof MutableRecipe)  {
+				if (recipe instanceof MutableCraftingRecipe mutableCraftingRecipe) {
+					return mutableCraftingRecipe.getGroup();
+				} else if (recipe instanceof MutableCookingRecipe mutableCookingRecipe) {
+					return mutableCookingRecipe.getGroup();
+				} else if (recipe instanceof MutableStonecuttingRecipe mutableStonecuttingRecipe) {
+					return mutableStonecuttingRecipe.getGroup();
+				}
+			} else {
+				if (recipe instanceof ShapedRecipe shapedRecipe) {
+					return shapedRecipe.getGroup();
+				} else if (recipe instanceof ShapelessRecipe shapelessRecipe) {
+					return shapelessRecipe.getGroup();
+				} else if (recipe instanceof CookingRecipe<?> cookingRecipe) {
+					return cookingRecipe.getGroup();
+				} else if (recipe instanceof StonecuttingRecipe stonecuttingRecipe) {
+					return stonecuttingRecipe.getGroup();
+				}
+			}
```

Comment:
what's going on here. Why isn't getGroup a method on MutableRecipe?
Why are you using specific getGroups? Why isn't it just CraftingRecipe#getGroup

Reply:
I forgot to note it, 1.19.4 -> 1.20 Dont have a Crafting Recipe class, thus instead of doing a check to see if what mc version it's running, I just did it this way

Reply:
Also, not all recipes have a `group` so I figured it would be better to have them in their respective classes, to somewhat match the behaviors of the Bukkit recipes

Reply:
Which ones don't have a group? It seems to me like they all do since it's in crafting, cooking, and stonecutting recipe interfaces

Reply:
> Which ones don't have a group? It seems to me like they all do since it's in crafting, cooking, and stonecutting recipe interfaces

The only recipe that doesn't is the Smithing recipe implmentations, in the future that will include merchant recipes (if those are added in this module)